### PR TITLE
Integrated wallet connection using wagmi

### PR DESCRIPTION
**Pull request by UsamaSaleem14**
Work in this PR:

1. Added wagmi in the app for wallet connection

image

image

image

### CODE REVIEW

File: src/components/Header/ConnectorsModal.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+          enterTo="opacity-100"
+          leave="ease-in duration-200"
+          leaveFrom="opacity-100"
+          leaveTo="opacity-0"
+        >
+          <div className={style.overlay}></div>
+        </Transition.Child>
+        <div className={style.dialogMain}>
+          <div className={style.dialogDiv}>
+            <Transition.Child
```
Comment (TBomberman) [inline src/components/Header/ConnectorsModal.tsx:53]: Line 51-59, how come GS does not require this transition.child element? Can we remove this?

File: src/components/Header/index.tsx [hunk 2]
```diff
@@ -56,13 +73,39 @@ const Header = () => {
...
-      <div className={style.connectBtnDiv}>
-        <div className={style.btnConnect}>
-          <div className={style.btnConnectContainer}>
-            <p>Connect Wallet</p>
+      {isConnected ? (
+        <div>
+          <UserWallet disconnectWallet={disconnect} />
+        </div>
+      ) : (
+        // <div
```
Comment (TBomberman) [inline src/components/Header/index.tsx]: Can these comments be removed?

**Comment by TBomberman** [comment]
Add chevron icon.

File: src/components/Header/index.tsx (hunk 2 above)
Comment (UsamaSaleem14) [reply src/components/Header/index.tsx]: Done

File: src/components/Header/ConnectorsModal.tsx (hunk 1 above)
Comment (UsamaSaleem14) [reply src/components/Header/ConnectorsModal.tsx:53]: Transition child as a transition behavior to the view. This looks good to human eye. If I remove this it looks like it jumps suddenly towards me. In my opinion we should keep transition child. Let me know about your thoughts

**Comment by TBomberman** [comment]
1. Build is broken.

File: src/components/Header/ConnectorsModal.tsx (hunk 1 above)
Comment (TBomberman) [reply src/components/Header/ConnectorsModal.tsx:53]: ok

**Comment by UsamaSaleem14** [comment]
> 1. Build is broken.

Fixed

File: package.json [hunk 3]
```diff
@@ -53,5 +53,8 @@
     "eslint-plugin-react-hooks": "^4.6.0",
     "prettier": "2.8.1",
     "tailwindcss": "^3.2.4"
+  },
+  "resolutions": {
+    "eth-sig-util": "3.0.1"
```
Comment (TBomberman) [inline package.json]: Why do we need this?

**Pull request merged**